### ISSUE TITLE
ElementFinder.prototype.clone

### ISSUE THREAD

**Issue by elgalu** [angular/protractor#1303]
I'm currently implementing a clone myself but I see my adhoc effort is about to be broken after #1296

It would be nice to have a clone() method for ElementFinder and ElementArrayFinder that also takes in consideration cloning the parents up the tree.

``` js
// Adhoc clone() recursive method
ElementFinder.prototype.clone = function() {
    var elm = this;
    var newElm;
    newElm = element(elm.locator());
    // Copy relevant properties
    newElm.opt_actionResult_ = elm.opt_actionResult_;
    newElm.opt_index_ = elm.opt_index_;
    // Clone parents
    var elmIterator = elm;
    if (elm.parentElementFinder_) {
        // TODO: Detect Circular parents to avoid infinite loop.
        newElm.parentElementFinder_ = elm.parentElementFinder_.clone();
    }
    return newElm;
};
```
